rm-datasource/src/main/java/io/seata/rm/datasource/undo/oracle/OracleUndoLogManager.java

```diff
@@ -67,34 +73,37 @@ public int deleteUndoLogByLogCreated(Date logCreated, int limitRows, Connection
     }
 
     @Override
-    protected void insertUndoLogWithNormal(String xid, long branchId, String rollbackCtx,
-                                                byte[] undoLogContent, Connection conn) throws SQLException {
-        insertUndoLog(xid, branchId,rollbackCtx, undoLogContent, State.Normal, conn);
+    protected void insertUndoLogWithNormal(String xid, long branchId, String rollbackCtx, byte[] undoLogContent,
+                                           CompressorType compressType, Connection conn) throws SQLException {
+        insertUndoLog(xid, branchId,rollbackCtx, undoLogContent, compressType, State.Normal, conn);
     }
 
     @Override
     protected byte[] getRollbackInfo(ResultSet rs) throws SQLException {
         Blob b = rs.getBlob(ClientTableColumnsName.UNDO_LOG_ROLLBACK_INFO);
         byte[] rollbackInfo = BlobUtils.blob2Bytes(b);
-        return rollbackInfo;
+        CompressorType compressType = CompressorType.getByCode(rs.getInt(ClientTableColumnsName.UNDO_LOG_COMPRESS_TYPE));
```

Comment:
Why not reduce the number of duplicate lines of code by putting decompression into AbstractUndoLogManager?

Reply:
done